### ISSUE TITLE
Error when fetching packages from private repos

### ISSUE THREAD

**Issue by kahmali** [DispatchMe/mgp#16]
I can fetch out of my public repos just fine, but when I try to access my private repos I get the error:
```bash
stream.js:94
      throw er; // Unhandled stream error in pipe.
            ^
Error: incorrect header check
    at Zlib._binding.onerror (zlib.js:295:17)
```

Is there something special that needs to be done when generating a token? Any specific permissions that need to be added?

**EDIT:** This is happening on my personal repos, not Dispatch.

**Comment by jperl** [comment]
We ran into this issue recently and realized that the code was not pushed yet. Please check if the commit is on github?

We should have better logging around this.

**Comment by aldeed** [comment]
Actually I just reproduced this with a private package of mine. The commit is definitely pushed to github this time. It will be helpful if we get the proper error logging added.

@jperl Do you know what permissions you selected when generating the token you  created?

**Comment by kahmali** [comment]
@jperl The commit I'm trying to fetch is pushed. Just verified.

**Comment by aldeed** [comment]
I submitted PR #18 with better logging around this. In my case, it's a 404 Not Found error. My hunch is that it might be because my organization has restricted access set up (have to manually approve new oauth apps). So personal access tokens are probably not an approved way of accessing the repo.

**Comment by jperl** [comment]
We are moving away from tarballs to using pure git. So this will be fixed in #19
